### PR TITLE
chore: stop using substra-tools base images

### PR DESCRIPTION
**Pull request by SdgJlbl**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ test: test-remote test-local
 test-remote: test-remote-sdk test-remote-workflows
 
 test-remote-sdk: pyclean
-	pytest tests -rs -v --durations=0 -m "not workflows" -n $(PARALLELISM) --log-level=INFO
+	pytest tests -x -rs -v --durations=0 -m "not workflows" -n $(PARALLELISM) --log-level=INFO
```
Comment (SdgJlbl) [inline Makefile:17]: TODO remove

File: tests/test_gpu.py [hunk 2]
```diff
@@ -11,8 +11,8 @@ def test_gpu(factory, client, org_idx, default_datasets, workers):
     nvidia_drivers = "nvidiacuda11.8.0-base-ubuntu22.04"
 
     # Need the base image, the minimal image does not have pip
-    dockerfile = f"""
-FROM ghcr.io/substra/substra-tools:latest-{nvidia_drivers}-python3.7
+    dockerfile = """
+FROM python:3.12-slim
```
Comment (thbcmlowk) [inline tests/test_gpu.py:15]: Isn't that slightly anticipated? 
```suggestion
FROM python:3.11-slim
```

File: substratest/settings.py [hunk 3]
```diff
@@ -13,13 +13,11 @@
 _DEFAULT_NETWORK_CONFIGURATION_PATH = os.path.join(_CURRENT_DIR, "../", "values.yaml")
 _SUBSTRA_TESTS_CONFIG_FILEPATH = os.getenv("SUBSTRA_TESTS_CONFIG_FILEPATH", _DEFAULT_NETWORK_CONFIGURATION_PATH)
 
-_DEFAULT_SUBSTRA_TOOLS_TAG_LOCAL = (
-    f"latest-nvidiacuda11.8.0-base-ubuntu22.04-python{sys.version_info.major}.{sys.version_info.minor}"
-)
-_DEFAULT_SUBSTRA_TOOLS_TAG_REMOTE = "latest"
+_DEFAULT_DOCKER_TAG_LOCAL = f"{sys.version_info.major}.{sys.version_info.minor}-slim"
+_DEFAULT_DOCKER_TAG_REMOTE = "3.11-slim"
```
Comment (guilhem-barthes) [inline substratest/settings.py:17]: Why do we need these two version, won't `_DEFAULT_DOCKER_TAG_LOCAL` return `3.11-slim` if started from the correct Python version? 

I can see that before, there was a difference in path between the two, but it looks like to me that we could use this opportunity to simplify the code

File: tests/test_gpu.py (hunk 2 above)
Comment (SdgJlbl) [reply tests/test_gpu.py:15]: Indeed, good catch

**Review by guilhem-barthes — APPROVED** [review]
🧹

**Pull request merged**